ql/pricingengines/credit/isdacdsengine.hpp

```diff
@@ -0,0 +1,149 @@
+            AccrualBias / ForwardsInCouponPeriod corresponding
+            to the standard model implementation C code are
+
+            prior 1.8.2    HalfDayBias / Flat
+            1.8.2          NoBias / Flat
+
+            The theoretical correct setting would be NoBias / Piecewise
+
+            Todo: Clarify in which version of the standard model
+            implementation C code the numerical problem of zero denominators
+            is solved and how exactly.
+        */
+
+        enum NumericalFix {
+            None,  // as in [1] footnote 26 (i.e. 10^{-50} is added to
+                   // denominators $f_i+h_i$$)
+            Taylor // as in [2] i.e. for $f_i+h_i < 10^{-4}$ a Taylor expansion
+                   // is used to avoid zero denominators
+        };
+
+        enum AccrualBias {
+            HalfDayBias, // as in [1] formula (50), second (error) term is
+                         // included
+            NoBias // as in [1], but second term in formula (50) is not included
+        };
+
+        enum ForwardsInCouponPeriod {
+            Flat, // as in [1], formula (52), second (error) term is included
+            Piecewise // as in [1], but second term in formula (52) is not
+                      // included
+        };
+
+        /*! Constructor where the client code is responsible for providing a
+            default curve and an interest rate curve compliant with the ISDA
+            specifications.
+
+            To be precisely consistent with the ISDA specification
+            QL_USE_INDEXED_COUPON
+            must not be defined. This is not checked in order not to
+            kill the engine completely in this case.
+
+            Furthermore, the ibor index in the swap rate helpers should not
+            provide the evaluation date's fixing.
+        */
+
+        IsdaCdsEngine(
+            const Handle<DefaultProbabilityTermStructure> &probability,
+            Real recoveryRate,
+            const Handle<YieldTermStructure> &discountCurve,
+            boost::optional<bool> includeSettlementDateFlows = boost::none,
+            const NumericalFix numericalFix = Taylor,
+            const AccrualBias accrualBias = NoBias,
+            const ForwardsInCouponPeriod forwardsInCouponPeriod = Piecewise);
+
+        /*! Constructor where the engine produces ISDA compliant interest and
+            credit curves. For the interest rate curve, rate helpers
+            compliant with [3] must be given. For the credit curve, the
+            discount curve handle in the helpers can be left empty
+            (since the rate curve is yet to be built). If more
+            than one helper is given, one should consider to convert each
+            one to an upfront helper (if it is given in conventional spread
+            flavour) to ensure precise consistency between the converter
+            pricing and full hazard term structure pricing */
+
+        IsdaCdsEngine(
```

Comment:
I'm not sure if I would add this constructor.  It's probably convenient, but I'm afraid it might start a trend 😄 .  I'd leave the construction of the curves to the user of the engine.

Reply:
This might also remove the need for `setDiscountCurve` in the helpers, if I'm not mistaken.